frontend/src/pages/CharacterSelectionPage/CharacterSelection.tsx

```diff
@@ -1,114 +1,87 @@
-            setHeroes(prevHeroes => prevHeroes.filter(hero => hero.id !== saveGameToDelete.heroId));
-        } catch (error) {
-            console.error('Error when deleting the save game or hero:', error);
-        }
-    };
 
     const onAddHero = (newHero: HeroDto) => {
-        setHeroes(prevHeroes => [...prevHeroes, newHero]);
+        setHeroes((prevHeroes) => [...prevHeroes, newHero]);
         setAddingHero(false);
         setShouldUpdate(true);
     };
+    const handleDeleteSaveGame = (saveId: string) => {
+        setSaveGames((prevSaveGames) =>
+            prevSaveGames.filter((game) => game.saveId !== saveId)
+        );
+        setHeroes((prevHeroes) => prevHeroes.filter((hero) => hero.id !== saveId));
+    };
 
-    const handleAddHeroClick = () => {
+    const handleAddNewHero = () => {
         setAddingHero(true);
     };
 
-    const handleUpdateHeroClick = (hero: HeroDto | undefined) => {
+    const handleUpdateHero = (hero: HeroDto | undefined) => {
         if (hero) {
             setUpdatingHero(hero);
             setUpdatingHeroDialogOpen(true);
         }
     };
 
     if (error) {
-        return <div className="error-message">Fehler: {error}</div>;
+        return <section className="error-message">Fehler: {error}</section>;
     }
 
     return (
-        <div className="character-selection-container">
+        <section style={{backgroundImage: `url(${backgroundImage})`}} className="character-selection-container">
+
             <h1>Saved Games</h1>
             <CharacterList
                 saveGames={saveGames}
                 heroes={heroes}
-                onDeleteSaveGame={onDeleteSaveGame}
-                onUpdateHeroClick={handleUpdateHeroClick}
+                onDeleteHero={handleDeleteSaveGame}
+                onDeleteSaveGame={handleDeleteSaveGame}
+                onUpdateHeroClick={handleUpdateHero}
             />
-            <button className="add-hero-button" onClick={handleAddHeroClick}>Neuen Held erstellen</button>
-            {addingHero && <AddHero onAddHero={onAddHero} setShouldUpdate={setShouldUpdate} />}
+            {addingHero && (
+                <AddHero
+                    onAddHero={onAddHero}
+                    setShouldUpdate={setShouldUpdate}
+                    onClose={() => setAddingHero(false)}
+                />
+            )}
             {updatingHeroDialogOpen && updatingHero && (
                 <UpdateHero
                     hero={updatingHero}
-                    onUpdateHero={updatedHero => {
-                        setHeroes(prevHeroes => prevHeroes.map(hero => (hero.id === updatedHero.id ? updatedHero : hero)));
+                    onUpdateHero={(updatedHero) => {
+                        setHeroes((prevHeroes) =>
+                            prevHeroes.map((hero) =>
+                                hero.id === updatedHero.id ? updatedHero : hero
+                            )
+                        );
```

Comment:
I don't get what is happening here. This need some refactoring for better understanding.